scipy/special/__init__.py

```diff
@@ -640,7 +641,7 @@
 from ._ufuncs import *
 
 from .basic import *
-from ._logsumexp import logsumexp
+from ._logsumexp import logsumexp, softmax
```

Comment:
Is it possible to rename this to `_expfuncs.py`?

Reply:
Since it is a private module, I'm not too worried about the name.  And once you realize that the softmax function is the gradient of `logsumexp`, it isn't so bad leaving the name as `_logsumexp.py`. 😃  That also suggests that a better name for `softmax` is something like `logsumexp_grad`, but `softmax` is what people will be looking for.

Reply:
> And once you realize that the softmax function is the gradient of logsumexp

This seems like an interesting enough observation that it could go into the notes section of the docs

Reply:
> This seems like an interesting enough observation that it could go into the notes section of the docs

Done (if somewhat tersely).